### PR TITLE
Bump version in master to 3.0

### PR DESCRIPTION
**Pull request by genlu**
All bug fixes for 2.9.0 should go in 2.9.x branch. I will go figure out how to setup code flow.

@jinujoseph @mavasani @sharwell @dotpaul

### CODE REVIEW

File: eng/Versions.props [hunk 1]
```diff
@@ -2,7 +2,7 @@
   <PropertyGroup>
     <MSBuildAllProjects>$(MSBuildAllProjects);$(MSBuildThisFileFullPath)</MSBuildAllProjects>
     
-    <VersionPrefix>2.9.1</VersionPrefix>
+    <VersionPrefix>3.0.0</VersionPrefix>
```
Comment (mavasani) [inline eng/Versions.props]: Our analyzer package version matches the version of referenced Microsoft.CodeAnalysis NuGet package. So this change should be accompanied by bumping up the CodeAnalysis version to 3.0.x

**Comment by genlu** [comment]
@ManishJayaswal Needed to change target framework to net472/netstandard20 due to roslyn change.

**Comment by genlu** [comment]
@sharwell Could you please take another look? Thanks

**Comment by sharwell** [comment]
@genlu This is blocked on https://github.com/dotnet/roslyn/issues/34151

**Pull request merged**